### PR TITLE
Add script to help with OCP certification

### PR DESCRIPTION
**Pull request by lwr20**
## Description

We aim to certify releases with RH Openshift.  This process includes running tests against all the images included in our operator for all the architectures that Openshift and Calico support.  This script makes that process less tedious.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: hack/certification/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+output-*
```
Comment (fasaxc) [inline hack/certification/.gitignore:1]: Best to keep the newline in this file, I think.

File: hack/certification/certify.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+# This script runs Openshift preflight against Calico images and optionally uploads the results to Openshift Connect for the certification process.
```
Comment (fasaxc) [inline hack/certification/certify.sh:2]: Please add the license/copyright header.

File: hack/certification/certify.sh [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+# In addition, the certification process also requires the following, which are NOT handled by this script:
+# - the operator metadata bundle (see the tigera/operator repo) to be submitted by PR
+#   to https://github.com/redhat-openshift-ecosystem/certified-operators
+# - running the Openshift End-to-End tests - we run them for CNI and CNV.  The results need to be passed to the Certification team for verification.
+
+set -x
```
Comment (fasaxc) [inline hack/certification/certify.sh:8]: Can we use `set -e` to fail on errors?  Or does that not make sense for this script?

File: hack/certification/certify.sh [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+# Redhat API key, needed for submitting the images
+RH_API_KEY="${RH_API_KEY:-""}"
+
+# SUBMIT controls whether this is a dry run or a "real" run.  Set to "--submit" to actually submit the images to the certification system.
+SUBMIT="${SUBMIT:-""}"
+
+podman pull quay.io/opdev/preflight:$PREFLIGHT_TAG
+
+declare -a LINUX_PLATFORMS=( \
```
Comment (fasaxc) [inline hack/certification/certify.sh:25]: I don't think you need the `\`s

File: hack/certification/certify.sh [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            -v "$PWD":/artifacts  \
+            quay.io/opdev/preflight:$PREFLIGHT_TAG check container ${ORG}/${IMAGE}:${VERSION} ${SUBMIT}
+        popd || exit
+    done
+}
+
+mkdir -p output-${CALICO_VERSION}
+pushd output-${CALICO_VERSION} || exit
+
+for IMAGE in "${!calico_image_project[@]}"; do \
```
Comment (fasaxc) [inline hack/certification/certify.sh:86]: No `\` needed

**Review by fasaxc — CHANGES_REQUESTED** [review]

File: hack/certification/certify.sh (hunk 3 above)
Comment (lwr20) [reply hack/certification/certify.sh:8]: Preflight exits with 1 for failed tests.  I really want it to continue and test the other images.  

I guess I could have added `set -ex` here and then `|| true` for the preflight tests only.  

But that's 90% of this script, so I felt it was simpler not to do that.

**Review by fasaxc — APPROVED** [review]

**Pull request merged**